### PR TITLE
tests and docstrings for null_driver and generic_driver

### PR DESCRIPTION
**Pull request by dbox-fnal**

### CODE REVIEW

File: src/null_driver.py [hunk 1]
```diff
@@ -15,16 +15,26 @@ class NullDriver(generic_driver.Driver):
...
         self._active_time = 0
         self._total_time = 0
-        self._bytes_transferred = 0L
+        self._bytes_transferred = 0
         self.verbose = 0
         
     def open(self, device=None, mode=None, retry_count=10):
+        """open the device
+            Args:
+                device(str): device name (optional)
```
Comment (renbauer-fnal) [inline src/null_driver.py]: IIRTC `device` is overwritten no matter what, right? We should rename this argument to `_` and note it is unused.

File: src/null_driver.py [hunk 2]
```diff
@@ -93,12 +155,26 @@ def read(self, buf, offset, nbytes):
...
+            int: number of bytes written
+                 -1 if IOError
+            
+        Raises:
+            ValueError if file not open for writing
+        """
         if self.mode != 1:
-            raise ValueError, "file not open for writing"
+            raise ValueError("file not open for writing")
         return nbytes
```
Comment (renbauer-fnal) [inline src/null_driver.py:177]: This function returns here. I think we can delete the rest of this code

**Review by renbauer-fnal — APPROVED** [review]
These look great! A couple small comments. Pre-approving as I am OOO next week. Please feel free to merge after addressing.

File: src/null_driver.py (hunk 1 above)
Comment (dbox-fnal) [reply src/null_driver.py]: I changed (optional) to (not used, but determined by mode), but opted to keep the var name as 'device' .

**Pull request merged**